drupal/build.sh

```diff
@@ -19,712 +19,695 @@ import tarfile
-				if os.path.isdir(fullpath):
-					self.notice("Removing old build " + f)
-					shutil.rmtree(fullpath)
-				elif os.path.isfile(fullpath):
-					self.notice("Removing old build archive " + f)
-					os.remove(fullpath)
-
-	# Purge current final build
-	def purge(self):
-		self.notice("Purging current build")
-		if self.hasExistingBuild():
-			self._wipe()
-
-	# Link
-	def link(self):
-		# Link and copy required files
-		self._link()
-		self._copy()
-		self.linked = True
-
-	# Finalize new build to be the final build
-	def finalize(self):
-		self.notice("Finalizing new build")
-		if os.path.isdir(self.final_build_dir):
-			self._unlink()
-			self._ensure_writable(self.final_build_dir)
-			os.rename(self.final_build_dir, self.final_build_dir_bak)
-		# Make sure linking has happened
-		if not self.linked:
-			self.link()
-		os.rename(self.temp_build_dir, self.final_build_dir)
-		if os.path.isdir(self.final_build_dir_bak):
-			shutil.rmtree(self.final_build_dir_bak, True)
-
-	# Print notice
-	def notice(self, *args):
-		print "\033[92m** BUILD NOTICE: \033[0m" + ' '.join(str(a) for a in args)
-
-	# Print errror
-	def error(self, *args):
-		print "\033[91m** BUILD ERROR: \033[0m" + ' '.join(str(a) for a in args)
-
-	# Print warning
-	def warning(self, *args):
-		print "\033[93m** BUILD WARNING: \033[0m" + ' '.join(str(a) for a in args)
-
-	# Run install
-	def install(self):
-		if not self._drush([
-			"--root=" + format(self.final_build_dir + self.drupal_subpath),
-			"site-install",
-			self.profile_name,
-			"install_configure_form.update_status_module='array(FALSE,FALSE)'"
-			"--account-name=admin",
-			"--account-pass=admin",
-			"--site-name=" + self.site_name,
-			"-y"
-		]):
-			raise BuildError("Install failed.");
-
-	# Update existing final build
-	def update(self):
-		if self._drush([
-			"--root=" + format(self.final_build_dir + self.drupal_subpath),
-			'updatedb',
-			'--y'
-		]):
-			self.notice("Update process completed")
-		else:
-			self.warning("Unable to update")
-
-	# Ask user for verification
-	def verify(self, text):
-		if text:
-			response = raw_input(text)
-		else:
-			response = raw_input("Type yes to verify that you know what you are doing: ")
-		if response.lower() != "yes":
-			raise BuildError("Cancelled by user")
-
-	# Execute a shell command
-	def shell(self, command):
-		if isinstance(command, list):
-			for step in command:
-				value = os.system(command) == 0
-				if not value:
-					return False
-			return True
-		else:
-			return os.system(command) == 0
-
-	# Execute a drush command
-	def drush_command(self, command):
-            drush_command = ['--root=' + format(self.final_build_dir + self.drupal_subpath)] + command.split(' ')
+    def __init__(self, settings):
+
+        self.composer = settings.get('composer', 'composer')
+        self.drush = settings.get('drush', 'drush')
+        self.type = settings.get('type', 'drush make')
+        self.drupal_version = settings.get('drupal_version', 'd7')
```

Comment:
Should this line be here? Or rather, is there a default `d8` value somewhere in settings for this branch?